javascript/javascript3/week1/city.js

```diff
@@ -0,0 +1,62 @@
+
+const apiKey = "06150efd3d284c694f2952a08be1740b";
+const main = document.getElementById('main');
+const form = document.getElementById('form');
+const search = document.getElementById('search');
+const button = document.getElementById('myBtn');
+const url = (city) => `https://api.openweathermap.org/data/2.5/weather?q=${city}&appid=${apiKey}`;
+async function getWeatherByLocation(city) {
+  const resp = await fetch(url(city), {
+    origin: "cros"
+  });
+  const respData = await resp.json();
+  addWeatherToPage(respData);
+}
+
+function addWeatherToPage(data) {
+  const temp = Ktoc(data.main.temp);
+  const speed = (data.wind.speed);
+  const sunrise = new Date((data.sys.sunrise) * 1000).toLocaleTimeString();
+  const sunset = new Date((data.sys.sunset) * 1000).toLocaleTimeString();
+  let latitude = (data.coord.lat);
+  let longitude = (data.coord.lon);
+  const weather = document.createElement('div')
+  weather.classList.add('weather');
+  weather.innerHTML = `
+      <h2><img src="https://openweathermap.org/img/wn/${data.weather[0].icon}@2x.png" /> ${temp}°C <img src="https://openweathermap.org/img/wn/${data.weather[0].icon}@2x.png" /></h2>  
+      <small>${data.weather[0].main}</small><br>  <small>WindSpeed: ${speed} m/s</small> <br> <small>Sunrise: ${sunrise}</small> <br> <small>Sunset: ${sunset}</small> <br> <small>Latitude: ${latitude}</small> <br> <small>Longitude: ${longitude}</small>
+      `
+  main.innerHTML = "";
+  main.appendChild(weather);
+}
+
+function Ktoc(K) {
+  return Math.floor(K - 273.15);            // convert the temperature from farenheit to celsius
+}
+form.addEventListener('submit', (e) => {
```

Comment:
This is a very good way of capturing form events, since it triggers regardless of how the user submits the form in their browser (return key, submit button etc.) 👍